code/modules/xenoarchaeology/xenoartifact_console.dm

```diff
@@ -0,0 +1,248 @@
+        if(action == "purchase_[S.unique_id]")
+            if(linked_inbox && budget.account_balance-S.price >= 0)
+                var/obj/item/xenoartifact/A = new (get_turf(linked_inbox.loc), S.difficulty)
+                var/datum/component/xenoartifact_pricing/X = A.GetComponent(/datum/component/xenoartifact_pricing)
+                if(!X)
+                    return
+                X.price = S.price
+                sellers -= S
+                budget.adjust_money(-1*S.price)
+                say("Purchase complete. [budget.account_balance] credits remaining in Research Budget")
+                addtimer(CALLBACK(src, .proc/generate_new_seller), (rand(1,5)*60) SECONDS)
+                return
+            else if(!linked_inbox)
+                say("Error. No linked hardware.")
+                return
+            else if(budget.account_balance-S.price < 0)
+                say("Error. Insufficient funds.")
+                return
+
+    if(action == "sell")
+        if(!linked_inbox)
+            say("Error. No linked hardware.")
+            return
+        var/info
+        var/final_price = 100
+        var/list/nearby = oview(1, linked_inbox)
+        for(var/obj/I in nearby)
+            var/avoidtimewaste = TRUE
+            for(var/datum/xenoartifactseller/buyer/B as() in buyers)//Check to avoid wasting time & to see if someone is actually buying that item.
+                if(istype(I, B.buying))
+                    avoidtimewaste = FALSE
+                    buyers -= B
+                    break
+            if(avoidtimewaste)
+                return
+            if(istype(I, /obj/item/xenoartifact)||istype(I, /obj/structure/xenoartifact)) //This and it's brother is pretty iffy
+                var/datum/component/xenoartifact_pricing/X = I.GetComponent(/datum/component/xenoartifact_pricing)
+                if(!X)
+                    return
+                final_price = X.modifier*X.price
+                if(final_price < 0) //No modulate?
+                    final_price = X.price*0.1
+                budget.adjust_money(final_price)
+                linked_techweb.add_point_type(TECHWEB_POINT_TYPE_DEFAULT, final_price*10)
+                info = "[I.name] sold at [station_time_timestamp()] for [final_price] credits, bought for [X.price]"
+                sold_artifacts += list(info)
+                say(info)
+                qdel(I)
+                addtimer(CALLBACK(src, .proc/generate_new_buyer), (rand(1,5)*60) SECONDS)
+                return
+            else    
+                final_price = 125*rand(0.3, 1.8) //This may be a point of conflict/balance
+                info = "[I] sold at [station_time_timestamp()] for [final_price]. No further information available."
+                sold_artifacts += list(info)
+                say(info)
+                qdel(I)
+                addtimer(CALLBACK(src, .proc/generate_new_buyer), (rand(1,5)*60) SECONDS)
+                return
```

Comment:
Wait, can't you sell literally anything (Including indestructible and rare items)?

Reply:
Only if someone is buying it but, buyers will only ever buy xenoartifacts. In the future, after I replace the purchase system, I'd like science to be able to occasionally have the opportunity to sell rare slime cores and similar.